src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-tutorial-advanced-dashboarding.mdx

```diff
@@ -0,0 +1,471 @@
+FROM Public_APICall 
+SINCE 1 day ago
+```
+<img
+    title="Null values"
+    alt="A screenshot showing the null values functionality"
+    src={queriesnrql2tutorial14}
+/>
+    </Step>
+    <Step>
+
+You've almost learned everything you need to help you advance your dashboarding! Next up, you'll learn how to use coercion.
+
+#### Coercion
+
+NRQL doesn't automatically apply coercion. This means we treat a float stored as a string as we would any other string, and you can't use them with mathematical functions like `sum()` or `average()`. To override this behavior, use `boolean()` or `numeric()` to convert arguments to boolean or numerical values. In this example, an `average()` function on "duration" provides no value since this attribute is a string. But if you convert the attribute to a number using `numeric(duration)`, you can use the `average()` function successfully.
+
+```sql
+SELECT average(numeric(duration)) AS 'Ensuring stored value is treated as numeric', average(duration) AS 'Non-Converted Attribute' 
+FROM Public_APICall 
+SINCE 1 day ago
+```
+<img
+    title="Coercion"
+    alt="A screenshot showing the coercion functionality"
+    src={queriesnrql2tutorial15}
+/>
+    </Step>
+    <Step>
+
+Another common example is `BOOLEAN` (TRUE or FALSE) values. These often incorrectly format as strings. When this happens, you can change how the source sends the data to make it a proper boolean. Or, you can use the `boolean()` function. The example query below returns the same result, but only because you use a value sent by the agent as a `BOOLEAN`. If your attribute was a string "TRUE", `boolean()` would convert it into a proper boolean format, allowing the query to run as intended.
+
+```sql
+SELECT count(boolean(sampleDataSet)), count(sampleDataSet)  
+FROM Public_APICall 
+SINCE 24 hours ago
+```
+<img
+    title="Coercion with boolean"
+    alt="A screenshot showing the coercion functionality with the boolean function"
+    src={queriesnrql2tutorial16}
+/>
+    </Step>
+    <Step>
+You can also convert boolean and numeric values to strings by using the `string()` function. With numeric values as floating-point numbers, you can use the optional `precision` argument to limit the number of decimal places for the string. This query returns the duration value as a string limited to three decimal places.
+
+```sql
+SELECT string(duration, precision: 3) 
+FROM Public_APICall 
+SINCE 24 hours ago
+```
+<img
+    title="Coercion with precision"
+    alt="A screenshot showing the coercion functionality with precision"
+    src={queriesnrql2tutorial17}
+/>
+    </Step>
+</Steps>
+
+Here we've given you the power to control your data formats and tell NRQL how you want it to act. NRQL operates in the manner we deem most logical, but if that doesn't suit your unique scenario, you can use the functions explored in this lesson to override those values. You have only one thing left to learn: string concatenation.
+
+### Use string concatenation
```

Comment:
Again, 'use' and a vague NRQL-specific functionality doesn't really direct customers to something valuable. They don't know what they don't know, so titling these with a general description of an outcome is better for all of these and worth taking the time to do